### ISSUE TITLE
Stop integration via Public API doesn't work

### ISSUE THREAD

**Issue by mkralik3** [syndesisio/syndesis#4789]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
I have a running integration. When I call
```
curl -k -L -H -v -X PUT "Content-Type: application/json" -H "SYNDESIS-XSRF-TOKEN: awesome" -H 'Authorization: Bearer <TOKEN>' https://public-syndesis.my-minishift.syndesis.io/api/v1/public/integrations/timer-to-log/deployments/stop 
```
It returns 
```
{"errorCode":403,"userMsg":"Given request is not acceptable","developerMsg":"Integration timer-to-log is not published"}
```

On the other hand, when the integration is Unpublished, it returns `204 No Content`. In my POV, it should return the error to notify a user that the integration is not published. 

![output](https://user-images.githubusercontent.com/16251792/53956354-af070880-40db-11e9-8c4f-4086e13b5cc0.gif)

**Comment by heiko-braun** [comment]
Is the resource name in the API maybe the pod name, rather than the common name?

**Comment by mkralik3** [comment]
Using the name of the integration works for another rest points from Public API. The name of the pod (e.g. i-timer-to-log-8-j5c8d) returns 404.

**Comment by heiko-braun** [comment]
ok, I was just guessing.

**Comment by heiko-braun** [comment]
Raising the priority acccording to @gaughan 's feedback (read: don't shoot the messenger)

**Comment by heiko-braun** [comment]
@dhirajsb Have you seen this?

**Comment by dhirajsb** [comment]
@heiko-braun I will resolve this issue today.

**Comment by dhirajsb** [comment]
@heiko-braun are blocker fixes still going to master only or is there a release branch now?

**Comment by heiko-braun** [comment]
Blockers need to go to 1.6.x

**Comment by dhirajsb** [comment]
This needs a little more work. The previous fix doesn't addess all integration states like deploying/errors, etc.

**Comment by mkralik3** [comment]
I noticed that it works only once after the integration is published via PublicAPI. When I stop integration and publish it again, I am not able to stop it again. The result is:
```
{"errorCode":403,"userMsg":"Given request is not acceptable","developerMsg":"Integration timer-to-log3 is not published"}
```
however, the `/state` point returns:
```
{"currentState":"Published"}
```
Step to reproduce:

1. Create simple integration and save it (doesn't publish it via UI)
2. Publish it (`curl --request POST .../timer-to-log/deployments`)
3. Stop it via Public API ( ` curl --request PUT .../timer-to-log/deployments/stop` )
4. Publish it again (`curl --request POST .../timer-to-log/deployments`)
5. Make sure that state is published (`curl --request GET .../timer-to-log/state`)
6. Stop it again ( ` curl --request PUT .../timer-to-log/deployments/stop` )

![output](https://user-images.githubusercontent.com/16251792/54289184-04925800-45a9-11e9-85e5-f7b21c8b8e34.gif)

**Comment by dhirajsb** [comment]
Ah, had one typo in the method name and ended up using the wrong method to lookup deployments. Somehow missed this in local testing. Opened a PR that should fix this issue.

**Comment by zregvart** [comment]
This is done right?

**Comment by heiko-braun** [comment]
It is done when https://github.com/syndesisio/syndesis/pull/4883 get's merged

**Comment by dhirajsb** [comment]
I opened another PR https://github.com/syndesisio/syndesis/pull/4892 to backport the test enhancement for 1.6.x

**Comment by heiko-braun** [comment]
@dhirajsb This issue got reopened due to the comments in https://github.com/syndesisio/syndesis/issues/4789#issuecomment-472459413. Have these been addressed with your recent PR?

**Comment by dhirajsb** [comment]
@heiko-braun yeah, that's the scenario tested in the new unit test I added. 
@mkralik3 confirmed it on Google Chat. Can you also add a comment here and close this issue?

**Comment by mkralik3** [comment]
Verified
